### PR TITLE
Clean ast core

### PR DESCRIPTION
**Pull request by privat**
A first attempt to PR something!

Some simple (I assume) and correct (I hope) cleaning and a small refactoring.

### CODE REVIEW

**Comment by Ducasse** [comment]
Nice. I will let marcus handle it.

**Review by MarcusDenker — APPROVED** [review]
Looks good, but we should wait for the CI to run once before merging

**Comment by MarcusDenker** [comment]
Failing tests not related:

[CI] Failing test: ReleaseTest>>testUndeclared https://github.com/pharo-project/pharo/issues/11670
[CI] Failing test: ProperMethodCategorizationTest>>testNoUncategorizedMethods https://github.com/pharo-project/pharo/issues/11669

**Pull request merged**
